src/windows_utils.rs

```diff
@@ -1,245 +1,83 @@
-    let str_icon = CString::new("Icon").unwrap();
-    let str_command = CString::new(format!("{} \"%1\"", neovide_path).as_bytes()).unwrap();
-    let str_description = CString::new("Open with Neovide").unwrap();
-    let str_neovide_path = CString::new(neovide_path.as_bytes()).unwrap();
-    unsafe {
-        if RegCreateKeyExA(
-            HKEY_CURRENT_USER,
-            str_registry_path.as_ptr(),
-            0,
-            null_mut(),
-            REG_OPTION_NON_VOLATILE,
-            KEY_WRITE,
-            null_mut(),
-            &mut registry_key,
-            null_mut(),
-        ) != 0
-        {
-            RegCloseKey(registry_key);
-            return false;
-        }
-        let registry_values = [
-            (
-                null(),
-                REG_SZ,
-                str_description.as_ptr() as *const u8,
-                str_description.to_bytes().len() + 1,
-            ),
-            (
-                str_icon.as_ptr(),
-                REG_SZ,
-                str_neovide_path.as_ptr() as *const u8,
-                str_neovide_path.to_bytes().len() + 1,
-            ),
-        ];
-        for &(key, keytype, value_ptr, size_in_bytes) in &registry_values {
-            RegSetValueExA(
-                registry_key,
-                key,
-                0,
-                keytype,
-                value_ptr,
-                size_in_bytes as u32,
-            );
-        }
-        RegCloseKey(registry_key);
-
-        if RegCreateKeyExA(
-            HKEY_CURRENT_USER,
-            str_registry_command_path.as_ptr(),
-            0,
-            null_mut(),
-            REG_OPTION_NON_VOLATILE,
-            KEY_WRITE,
-            null_mut(),
-            &mut registry_key,
-            null_mut(),
-        ) != 0
-        {
-            return false;
-        }
-        let registry_values = [(
-            null(),
-            REG_SZ,
-            str_command.as_ptr() as *const u8,
-            str_command.to_bytes().len() + 1,
-        )];
-        for &(key, keytype, value_ptr, size_in_bytes) in &registry_values {
-            RegSetValueExA(
-                registry_key,
-                key,
-                0,
-                keytype,
-                value_ptr,
-                size_in_bytes as u32,
-            );
-        }
-        RegCloseKey(registry_key);
-    }
-    true
+fn register_rightclick_file() -> Result<()> {
+    let neovide_path = get_neovide_path();
+    let neovide_description = "Open with Neovide";
+    let neovide_command = format!("{} \"%1\"", neovide_path);
+
+    let key = CURRENT_USER.create(REGISTRY_PATH_FOLDER)?;
+    key.set_string("", neovide_description)?;
+    key.set_string("Icon", &neovide_path)?;
+
+    let key = CURRENT_USER.create(REGISTRY_PATH_FOLDER_COMMAND)?;
+    key.set_string("", &neovide_command)?;
+
+    Ok(())
 }
 
 pub fn register_right_click() {
-    if unregister_rightclick() {
+    if unregister_rightclick().is_ok() {
```

Comment:
I think this check can be removed because this check seems redundant. That is, registering keys don't need to call `unregister_rightclick()`. It can be called when removing keys from registry.

Reply:
> This looks a bit strange. I suggest that you still check for `is_err` here but add a check in `unregister_rightclick` so that it returns ok when there's no previous entry.

I think this check can be removed because this check seems redundant. That is, registering keys don't need to call `unregister_rightclick()`. It can be called when removing keys from registry.

Reply:
> This looks a bit strange. I suggest that you still check for `is_err` here but add a check in `unregister_rightclick` so that it returns ok when there's no previous entry.

I think this check can be removed because this check seems redundant. That is, registering keys don't need to call `unregister_rightclick()`. It can only be called when removing keys from registry.

Reply:
> This looks a bit strange. I suggest that you still check for `is_err` here but add a check in `unregister_rightclick` so that it returns ok when there's no previous entry.

I think maybe this check can be removed because it seems doesn't matter that users repeatedly register keys. In fact, in the original code, keys will always be written in registry again whether the keys are registered before, because the way that check whether the keys are registered **ONLY** tells user that some keys are registered before but doesn't avoid calling `register_rightclick_directory()` and `register_rightclick_file()`.

Reply:
Yes, that seems like a good way to do it.